ballsdex/packages/players/cog.py

```diff
@@ -134,6 +157,258 @@ async def delete(self, interaction: discord.Interaction):
+            await interaction.followup.send("You cannot block yourself.", ephemeral=True)
+            return
+        if user.bot:
+            await interaction.followup.send("You cannot block a bot.", ephemeral=True)
+            return
+
+        blocked = await player1.is_blocked(player2)
+        if blocked:
+            await interaction.followup.send("You have already blocked this user.", ephemeral=True)
+            return
+
+        friended = await player1.is_friend(player2)
+        if friended:
+            view = ConfirmChoiceView(interaction)
+            await interaction.followup.send(
+                "This user is your friend, are you sure you want to block them?",
+                view=view,
+                ephemeral=True,
+            )
+            await view.wait()
+
+            if not view.value:
+                return
+            else:
+                await Friendship.filter(
+                    (Q(player1=player1) & Q(player2=player2))
+                    | (Q(player1=player2) & Q(player2=player1))
+                ).delete()
+
+        await Block.create(player1=player1, player2=player2)
+        await interaction.followup.send(f"You have now blocked {user.name}.", ephemeral=True)
+
+    @blocked.command(name="remove")
+    async def block_remove(self, interaction: discord.Interaction, user: discord.User):
+        """
+        Unblock a user.
+
+        Parameters
+        ----------
+        user: discord.User
+            The user you want to unblock.
+        """
+        player1, _ = await PlayerModel.get_or_create(discord_id=interaction.user.id)
+        player2, _ = await PlayerModel.get_or_create(discord_id=user.id)
+
+        if player1 == player2:
+            await interaction.response.send_message("You cannot unblock yourself.", ephemeral=True)
+            return
+        if user.bot:
+            await interaction.response.send_message("You cannot unblock a bot.", ephemeral=True)
+            return
+
+        blocked = await player1.is_blocked(player2)
+
+        if not blocked:
+            await interaction.response.send_message("This user isn't blocked.", ephemeral=True)
+            return
+        else:
+            await Block.filter(
+                (Q(player1=player1) & Q(player2=player2))
+                | (Q(player1=player2) & Q(player2=player1))
+            ).delete()
+            await interaction.response.send_message(
+                f"{user.name} has been unblocked.", ephemeral=True
+            )
+
+    @blocked.command(name="list")
+    async def blocked_list(self, interaction: discord.Interaction["BallsDexBot"]):
+        """
+        View all the users you have blocked.
+        """
+        player, _ = await PlayerModel.get_or_create(discord_id=interaction.user.id)
+
+        blocked_relations = (
+            await Block.filter(Q(player1=player) | Q(player2=player))
+            .select_related("player1", "player2")
+            .all()
+        )
```

Comment:
This query also shows if a user has you blocked, it should be changed for just 
```suggestion
        blocked_relations = (
            await Block.filter(player1=player)
            .select_related("player1", "player2")
            .all()
        )
```
this way its only people that player1 has blocked, aka the person invoking